cloud/pkg/cloudhub/channelq/channelq.go

```diff
@@ -108,14 +117,36 @@ func (q *ChannelMessageQueue) addMessageToQueue(nodeID string, msg *beehiveModel
 		// the version stored in the database
 		if !exist {
 			resourceNamespace, _ := edgemessagelayer.GetNamespace(*msg)
+			resourceName, _ := edgemessagelayer.GetResourceName(*msg)
 			resourceUID, err := GetMessageUID(*msg)
+			objectSyncName := synccontroller.BuildObjectSyncName(nodeID, resourceUID)
 			if err != nil {
 				klog.Errorf("fail to get message UID for message: %s", msg.Header.ID)
 				return
 			}
-			objectSync, err := q.objectSyncLister.ObjectSyncs(resourceNamespace).Get(synccontroller.BuildObjectSyncName(nodeID, resourceUID))
+			objectSync, err := q.objectSyncLister.ObjectSyncs(resourceNamespace).Get(objectSyncName)
 			if err == nil && objectSync.Status.ObjectResourceVersion != "" && synccontroller.CompareResourceVersion(msg.GetResourceVersion(), objectSync.Status.ObjectResourceVersion) <= 0 {
 				return
+			} else if err != nil && apierrors.IsNotFound(err) {
+				objectSync := &v1alpha1.ObjectSync{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: objectSyncName,
+					},
+					Spec: v1alpha1.ObjectSyncSpec{
+						ObjectAPIVersion: util.GetMessageAPIVerison(msg),
+						ObjectKind:       util.GetMessageResourceType(msg),
+						ObjectName:       resourceName,
+					},
+				}
+				objectSyncStatus, err := q.crdClient.ReliablesyncsV1alpha1().ObjectSyncs(resourceNamespace).Create(context.Background(), objectSync, metav1.CreateOptions{})
+				if err != nil {
+					klog.Errorf("Failed to create objectSync: %s, err: %v", objectSyncName, err)
+					return
+				}
+				objectSyncStatus.Status.ObjectResourceVersion = "0"
+				if _, err := q.crdClient.ReliablesyncsV1alpha1().ObjectSyncs(resourceNamespace).UpdateStatus(context.Background(), objectSyncStatus, metav1.UpdateOptions{}); err != nil {
+					klog.Errorf("Failed to update objectSync: %s, err: %v", objectSyncName, err)
+				}
 			}
 		}
```

Comment:
We shouldn't need to judge `exists` again, we can use `else` directly

Reply:
Here in line 146-149, I just update the ObjectSync's Status without judging `exists`. Can you say it more clearly?

Reply:
Sorry, I mean, there are `if !exist` in line 118 and `if exist` in line 154 in the code, can we merge them with `else`?

Reply:
Ok, I know.